modules/phase_field/include/materials/ComputeGBMisorientationType.h

```diff
@@ -0,0 +1,58 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+#include "GrainTracker.h"
+#include "EBSDReader.h"
+
+/**
+ * Visualize the location of grain boundaries in a polycrystalline simulation.
+ */
+class ComputeGBMisorientationType : public Material
+{
+public:
+  static InputParameters validParams();
+
+  ComputeGBMisorientationType(const InputParameters & parameters);
+
+protected:
+  /// Necessary override. This is where the property values are set.
+  virtual void computeQpProperties() override;
+
+  // Function to read the Misorientation angle form a file
+  void readFile();
+  // Function to output total line number of Misorientation angle file
+  virtual unsigned int getTotalLineNum() const;
+  // Function to output specific line number in Misorientation angle file
```

Comment:
this function does not output anything. better write "obtain line number for a given grain pair". 

Also please use parameter names in the header for self documentation purposes.

```
unsigned int getLineNum(unsigned int grain_i, unsigned int grain_j)
```